### PR TITLE
Use rp2040-hal in all example (possibly through their bsp)

### PR DESCRIPTION
**Pull request by ithinuel**
Some of the examples were using the cortex_m_rt::entry method which
misses the device specific spinlock re-initialisation.

This commits makes the usage more consistent by using rp2040_hal exported
macro as the only `entry` method used across examples.

### CODE REVIEW

**Comment by jannic** [comment]
`rp2040-hal/Cargo.toml.orig` was accidentally added to this merge request.

File: boards/arduino_nano_connect/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Changelog
```
Comment (jannic) [inline boards/arduino_nano_connect/CHANGELOG.md:1]: I'm not sure if changelogs for individual board crates are really useful. Updating them will be a lot of work, and in most cases they will all contain the same entries. IIRC there was a discussion about this in the past, but I don't remember where and what the conclusion was.

**Comment by ithinuel** [comment]
With this extra commit, it should no longer happen

**Review by jannic — CHANGES_REQUESTED** [review]
I like the idea to use `rp2040_hal::entry` in the examples. They provide a better default.

**Review by jannic — APPROVED** [review]
The previous "request changes" was due to the `.orig` files, which you already removed in the meantime.
Comments regardings changelogs and the way the entry macro is called are just ideas, not blockers.

**Pull request merged**